pkg/util/error.go

```diff
@@ -10,6 +12,25 @@ import (
 	"k8s.io/client-go/discovery"
 )
 
+const (
+	NoSuchHostMsg = "no such host"
+)
+
+func IsNoSuchHost(err error, lgr func(string)) bool {
```

Comment:
I dont like how this has been implemented with the logger. In the team we agreed on to avoid passing logger from function to function. And here you even had the need to write a wrapper function.

Normally, we try to just pass `ctx context.Context` and then use `logf.FromContext`, in this way you could also avoid introducing `"github.com/go-logr/logr"` into the `purge_controller.go`.